File: LeetCode/1678. Goal Parser Interpretation/solution.cpp
/**
 * Runtime: 4 ms
 * Memory Usage: 6 MB 
 */
class Solution {
public:
    string interpret(string command) {
        string ans;
        for (int i = 0; i < command.size(); ++ i) {
            char c = command[i];
            if (c == '(') {
                if (command[i + 1] == ')') {
                    ans += 'o';
                    ++ i;
                }
            } else {
                if (c == ')') continue;
                ans += c;
            }
        }
        return ans;
    }
};

File: LeetCode/1678. Goal Parser Interpretation/solution.py
"""
Runtime: 32 ms
Memory Usage: 14.9 MB
"""
class Solution:
    def interpret(self, command: str) -> str:
        ans = []
        for i, c in enumerate(command):
            if c == 'G':
                ans.append(c)
            elif c == '(':
                ans.append('o' if command[i + 1] == ')' else 'al')
        return ''.join(ans)
    